kernel/aster-nix/src/syscall/mod.rs

```diff
@@ -231,10 +359,22 @@ impl SyscallArgument {
     }
 }
 
-pub fn handle_syscall(context: &mut UserContext) {
+/// The information of current process, posix thread, thread and task.
+pub struct CurrentInfo<'a, 'b, 'c, 'd> {
+    pub process: &'a Process,
+    pub posix_thread: &'b PosixThread,
+    pub thread: &'c Thread,
+    pub task: &'d Task,
+}
```

Comment:
The task may be removed. In theory, syscalls will not try to get info from `Task`.

Reply:
Now they would, since I optimized `CurrentUserSpace::get` also.